src/core/service.c

```diff
@@ -1251,23 +1265,81 @@ static int service_coldplug(Unit *u) {
         return 0;
 }
 
+static int connect_to_socket(Service *s, OpenFile *f, int o_path_fd, int *ret) {
+        union sockaddr_union addr;
+        socklen_t sa_len;
+        int socket_types[] = { SOCK_DGRAM, SOCK_STREAM, SOCK_SEQPACKET };
+        int r;
+
+        r = sockaddr_un_set_path(&addr.un, FORMAT_PROC_FD_PATH(o_path_fd));
+        if (r < 0) {
+                return log_unit_error_errno(UNIT(s), r, "Failed to fill sockaddr for %s: %m", f->path);
+        }
+
+        sa_len = r;
+
+        for (size_t i = 0; i < ELEMENTSOF(socket_types); i++) {
+                _cleanup_close_ int fd = -1;
+
+                fd = socket(AF_UNIX, socket_types[i] | SOCK_CLOEXEC, 0);
+                if (fd < 0) {
+                        return log_unit_error_errno(
+                                        UNIT(s), errno, "Failed to create socket for %s: %m", f->path);
+                }
+
+                if (connect(fd, &addr.sa, sa_len) >= 0) {
```

Comment:
We shouldn't ignore every error here. I think we'll only want to ignore the error if its `EPROTOTYPE`, which indicates we're trying to connect to the wrong kind of socket.

Reply:
Also, we should use `connect_unix_path()` here instead of `connect()`, to automatically deal with long paths that can't be handled via raw `connect()` calls

Reply:
Are you sure that I need to use `connect_unix_path`? The path should will  be something like "/proc/self/fd/{FD}" so we should not have a long path issue

Reply:
The error return is not `EPROTOTYPE`, it's `EPERM`. So I ignore this one.